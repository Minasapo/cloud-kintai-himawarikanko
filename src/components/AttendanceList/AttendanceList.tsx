import {
  Box,
  Breadcrumbs,
  LinearProgress,
  Stack,
  styled,
  Typography,
} from "@mui/material";
import { DatePicker } from "@mui/x-date-pickers";
import { Logger } from "aws-amplify";
import dayjs from "dayjs";
import { useContext, useEffect, useState } from "react";
import { Link, useNavigate } from "react-router-dom";

import { calcTotalRestTime } from "@/pages/AttendanceEdit/DesktopEditor/RestTimeItem/RestTimeInput/RestTimeInput";
import { calcTotalWorkTime } from "@/pages/AttendanceEdit/DesktopEditor/WorkTimeInput/WorkTimeInput";

import { Staff } from "../../API";
import { useAppDispatchV2 } from "../../app/hooks";
import * as MESSAGE_CODE from "../../errors";
import useAttendances from "../../hooks/useAttendances/useAttendances";
import useCompanyHolidayCalendars from "../../hooks/useCompanyHolidayCalendars/useCompanyHolidayCalendars";
import useHolidayCalendars from "../../hooks/useHolidayCalendars/useHolidayCalendars";
import fetchStaff from "../../hooks/useStaff/fetchStaff";
import { AuthContext } from "../../Layout";
import { setSnackbarError } from "../../lib/reducers/snackbarReducer";
import Title from "../Title/Title";
import DesktopList from "./DesktopList";
import MobileList from "./MobileList/MobileList";

const DescriptionTypography = styled(Typography)(({ theme }) => ({
  padding: "0px 40px",
  [theme.breakpoints.down("md")]: {
    padding: "0px 10px",
  },
}));

export default function AttendanceTable() {
  const { cognitoUser } = useContext(AuthContext);
  const dispatch = useAppDispatchV2();
  const navigate = useNavigate();
  const { attendances, getAttendances } = useAttendances();
  const {
    holidayCalendars,
    loading: holidayCalendarLoading,
    error: holidayCalendarError,
  } = useHolidayCalendars();
  const {
    companyHolidayCalendars,
    loading: companyHolidayCalendarLoading,
    error: companyHolidayCalendarError,
  } = useCompanyHolidayCalendars();

  const [staff, setStaff] = useState<Staff | null | undefined>(undefined);
  const [totalTime, setTotalTime] = useState<number>(0);

  const logger = new Logger(
    "AttendanceList",
    import.meta.env.DEV ? "DEBUG" : "ERROR"
  );

  useEffect(() => {
    if (!cognitoUser) return;

    getAttendances(cognitoUser.id).catch((error) => {
      logger.debug(error);
      dispatch(setSnackbarError(MESSAGE_CODE.E02001));
    });

    fetchStaff(cognitoUser.id)
      .then((res) => {
        setStaff(res);
      })
      .catch((error) => {
        logger.debug(error);
        dispatch(setSnackbarError(MESSAGE_CODE.E00001));
      });
  }, [cognitoUser]);

  useEffect(() => {
    const totalWorkTime = attendances.reduce((acc, attendance) => {
      if (!attendance.startTime || !attendance.endTime) return acc;

      const workTime = calcTotalWorkTime(
        attendance.startTime,
        attendance.endTime
      );
      return acc + workTime;
    }, 0);

    const totalRestTime = attendances.reduce((acc, attendance) => {
      if (!attendance.rests) return acc;

      const restTime = attendance.rests
        .filter((item): item is NonNullable<typeof item> => !!item)
        .reduce((acc, rest) => {
          if (!rest.startTime || !rest.endTime) return acc;

          return acc + calcTotalRestTime(rest.startTime, rest.endTime);
        }, 0);

      return acc + restTime;
    }, 0);

    setTotalTime(totalWorkTime - totalRestTime);
  }, [attendances]);

  if (holidayCalendarLoading || companyHolidayCalendarLoading) {
    return <LinearProgress />;
  }

  if (holidayCalendarError || companyHolidayCalendarError) {
    dispatch(setSnackbarError(MESSAGE_CODE.E00001));
    return null;
  }

  return (
    <Stack spacing={2}>
      <Box>
        <Breadcrumbs>
          <Link to="/" color="inherit">
            TOP
          </Link>
          <Typography color="text.primary">勤怠一覧</Typography>
        </Breadcrumbs>
      </Box>
      <Box>
        <Title>{`勤怠一覧(${totalTime.toFixed(1)}h)`}</Title>
      </Box>
      <DescriptionTypography variant="body1">
        今日から30日前までの勤怠情報を表示しています
      </DescriptionTypography>
      <Box
        sx={{
          pl: {
            md: 5,
          },
        }}
      >
        <DatePicker
          value={dayjs()}
          format="YYYY/MM/DD"
          label="日付を指定して移動"
          slotProps={{
            textField: { size: "small" },
          }}
          onChange={(date) => {
            if (date) {
              navigate(`/attendance/${date.format("YYYYMMDD")}/edit`);
            }
          }}
        />
      </Box>
      <DesktopList
        attendances={attendances}
        holidayCalendars={holidayCalendars}
        companyHolidayCalendars={companyHolidayCalendars}
        navigate={navigate}
        staff={staff}
      />
      <MobileList
        attendances={attendances}
        holidayCalendars={holidayCalendars}
        companyHolidayCalendars={companyHolidayCalendars}
        staff={staff}
      />
    </Stack>
  );
}
