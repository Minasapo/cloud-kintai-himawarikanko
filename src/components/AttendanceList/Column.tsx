import EditIcon from "@mui/icons-material/Edit";
import {
  GridActionsCellItem,
  GridColDef,
  GridRowParams,
  GridValueGetterParams,
} from "@mui/x-data-grid";
import dayjs from "dayjs";
import { NavigateFunction } from "react-router-dom";

import {
  Attendance,
  CompanyHolidayCalendar,
  HolidayCalendar,
  Rest,
} from "../../API";
import {
  calcRestTotalTime,
  calcWorkTimeTotal,
  judgeStatus,
  makeRemarks,
  makeWorkDate,
  makeWorkEndTime,
  makeWorkStartTime,
} from "./common";

export interface DataGridProps {
  id: Attendance["id"];
  workDate: Attendance["workDate"];
  startTime: Attendance["startTime"];
  endTime: Attendance["endTime"];
  remarks: Attendance["remarks"];
  paidHolidayFlag: Attendance["paidHolidayFlag"];
  changeRequests: Attendance["changeRequests"];
  rests: Rest[];
}

export default function GetColumns(
  holidayCalendars: HolidayCalendar[],
  companyHolidayCalendars: CompanyHolidayCalendar[],
  navigate: NavigateFunction
): GridColDef[] {
  return [
    {
      field: "status",
      headerName: "ステータス",
      align: "center",
      valueGetter: (params: GridValueGetterParams<DataGridProps>) => {
        const {
          workDate,
          startTime,
          endTime,
          paidHolidayFlag,
          changeRequests,
        } = params.row;

        return judgeStatus(
          workDate,
          startTime,
          endTime,
          holidayCalendars,
          companyHolidayCalendars,
          paidHolidayFlag,
          changeRequests
        );
      },
    },
    {
      field: "workDate",
      headerName: "日付",
      align: "right",
      sortable: false,
      headerAlign: "center",
      valueGetter: (params: GridValueGetterParams<DataGridProps>) => {
        const { workDate } = params.row;
        return makeWorkDate(workDate, holidayCalendars);
      },
    },
    {
      field: "startTime",
      headerName: "出勤時刻",
      align: "right",
      sortable: false,
      headerAlign: "center",
      valueGetter: (params: GridValueGetterParams<DataGridProps>) => {
        const { startTime, paidHolidayFlag } = params.row;
        return makeWorkStartTime(startTime, paidHolidayFlag);
      },
    },
    {
      field: "endTime",
      headerName: "退勤時刻",
      align: "right",
      sortable: false,
      headerAlign: "center",
      valueGetter: (params: GridValueGetterParams<DataGridProps>) => {
        const { endTime, paidHolidayFlag } = params.row;
        return makeWorkEndTime(endTime, paidHolidayFlag);
      },
    },
    {
      field: "restTimeTotal",
      headerName: "休憩時間",
      align: "right",
      sortable: false,
      headerAlign: "center",
      valueGetter: (params: GridValueGetterParams<DataGridProps>) => {
        const { startTime, endTime, paidHolidayFlag, rests } = params.row;
        return calcRestTotalTime(startTime, endTime, paidHolidayFlag, rests);
      },
    },
    {
      field: "workTimeTotal",
      headerName: "勤務時間",
      align: "right",
      sortable: false,
      headerAlign: "center",
      valueGetter: (params: GridValueGetterParams<DataGridProps>) => {
        const { startTime, endTime, paidHolidayFlag, rests } = params.row;
        return calcWorkTimeTotal(
          params.row.workDate,
          startTime,
          endTime,
          paidHolidayFlag,
          rests
        );
      },
    },
    {
      field: "summary",
      headerName: "摘要",
      align: "left",
      sortable: false,
      width: 300,
      headerAlign: "center",
      valueGetter: (params: GridValueGetterParams<DataGridProps>) => {
        const { workDate, paidHolidayFlag, remarks } = params.row;
        return makeRemarks(
          workDate,
          paidHolidayFlag,
          remarks,
          holidayCalendars,
          companyHolidayCalendars
        );
      },
    },
    {
      field: "actions",
      headerName: "操作",
      type: "actions",
      sortable: false,
      getActions: (params: GridRowParams<DataGridProps>) => [
        <GridActionsCellItem
          key="edit"
          label="編集"
          icon={<EditIcon />}
          onClick={() => {
            const workDate = dayjs(params.row.workDate).format("YYYYMMDD");
            navigate(`/attendance/${workDate}/edit`);
          }}
        />,
      ],
    },
  ];
}
