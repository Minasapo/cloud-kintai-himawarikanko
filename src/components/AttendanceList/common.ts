import dayjs from "dayjs";

import {
  Attendance,
  CompanyHolidayCalendar,
  HolidayCalendar,
  Rest,
  Staff,
} from "../../API";
import getDayOfWeek from "./getDayOfWeek";

// --------------------------------------------------
//  日付
// --------------------------------------------------
export function makeWorkDate(
  workDate: Attendance["workDate"],
  holidayCalendars: HolidayCalendar[]
) {
  const date = dayjs(workDate);
  const isHoliday = holidayCalendars?.find(
    ({ holidayDate }) => holidayDate === workDate
  );
  const dayOfWeek = isHoliday ? "祝" : getDayOfWeek(workDate);
  return `${date.format("M/D")}(${dayOfWeek})`;
}

// --------------------------------------------------
//  勤務開始時間
// --------------------------------------------------
export function makeWorkStartTime(
  startTime: Attendance["startTime"],
  paidHolidayFlag: Attendance["paidHolidayFlag"]
) {
  if (paidHolidayFlag) return "09:00";
  return !startTime ? "" : dayjs(startTime).format("HH:mm");
}

// --------------------------------------------------
//  勤務終了時間
// --------------------------------------------------
export function makeWorkEndTime(
  endTime: Attendance["endTime"],
  paidHolidayFlag: Attendance["paidHolidayFlag"]
) {
  if (paidHolidayFlag) return "18:00";
  return !endTime ? "" : dayjs(endTime).format("HH:mm");
}

// --------------------------------------------------
//  休憩時間の合計
// --------------------------------------------------
export function calcRestTotalTime(
  workStartTime: Attendance["startTime"],
  workEndTime: Attendance["endTime"],
  paidHolidayFlag: Attendance["paidHolidayFlag"],
  rests: Rest[]
) {
  if (paidHolidayFlag) return "01:00";
  if (!workStartTime && !workEndTime) return "";
  if (rests.length === 0) return "00:00";

  const restTimeTotal = rests.reduce((acc, cur) => {
    const { startTime, endTime } = cur;
    if (!startTime || !endTime) return acc;

    const restTimeDiff = dayjs(endTime).diff(dayjs(startTime), "minute");
    return acc + restTimeDiff;
  }, 0);

  const totalTime = Math.floor(restTimeTotal / 60);
  const formatHour = totalTime.toString().padStart(2, "0");
  const formatMinute = (restTimeTotal % 60).toString().padStart(2, "0");
  return `${formatHour}:${formatMinute}`;
}

// --------------------------------------------------
//  勤務時間の合計
// --------------------------------------------------
export function calcWorkTimeTotal(
  workDate: Attendance["workDate"],
  startTime: Attendance["startTime"],
  endTime: Attendance["endTime"],
  paidHolidayFlag: Attendance["paidHolidayFlag"],
  rests: Rest[]
) {
  if (paidHolidayFlag) return "08:00";
  if (!startTime && !endTime) return "";

  if (!startTime || !endTime) {
    const todayStr = dayjs().format("YYYY-MM-DD");
    return todayStr === workDate ? "(計算中)" : "--:--";
  }

  const workTimeDiff = dayjs(endTime).diff(dayjs(startTime), "minute");

  const restTimeTotal = rests.reduce((acc, cur) => {
    const { startTime: restStartTime, endTime: restEndTime } = cur;
    if (!restStartTime || !restEndTime) return acc;

    const restStart = dayjs(restStartTime);
    const restEnd = dayjs(restEndTime);
    const restTime = restEnd.diff(restStart, "minute");
    return acc + restTime;
  }, 0);

  const workTimeTotal = workTimeDiff - restTimeTotal;
  const formatHour = Math.floor(workTimeTotal / 60)
    .toString()
    .padStart(2, "0");
  const formatMinute = (workTimeTotal % 60).toString().padStart(2, "0");
  return `${formatHour}:${formatMinute}`;
}

// --------------------------------------------------
//  摘要
// --------------------------------------------------
export function makeRemarks(
  workDate: Attendance["workDate"],
  paidHolidayFlag: Attendance["paidHolidayFlag"],
  remarks: Attendance["remarks"],
  holidayCalendars: HolidayCalendar[],
  companyHolidayCalendars: CompanyHolidayCalendar[]
) {
  const isHoliday = holidayCalendars?.find(
    ({ holidayDate }) => holidayDate === workDate
  );

  const isCompanyHoliday = companyHolidayCalendars?.find(
    ({ holidayDate }) => holidayDate === workDate
  );

  const summaryMessage = [];
  if (paidHolidayFlag) summaryMessage.push("有給休暇");
  if (isHoliday) summaryMessage.push(isHoliday.name);
  if (isCompanyHoliday) summaryMessage.push(isCompanyHoliday.name);
  if (remarks) summaryMessage.push(remarks);
  return summaryMessage.join(" ");
}

// --------------------------------------------------
//  ステータス
// --------------------------------------------------
export function judgeStatus(
  workDate: Attendance["workDate"],
  startTime: Attendance["startTime"],
  endTime: Attendance["endTime"],
  holidayCalendars: HolidayCalendar[],
  companyHolidayCalendars: CompanyHolidayCalendar[],
  paidHolidayFlag: Attendance["paidHolidayFlag"],
  changeRequests: Attendance["changeRequests"],
  staff: Staff | null | undefined
) {
  if (
    staff?.usageStartDate &&
    dayjs(staff.usageStartDate).isAfter(dayjs(workDate), "date")
  ) {
    return "";
  }

  if (paidHolidayFlag) return "OK";

  const today = dayjs().format("YYYY-MM-DD");
  const dayOfWeek = getDayOfWeek(workDate);
  const isHoliday = holidayCalendars?.find(
    (holiday) => holiday.holidayDate === workDate
  );
  const isCompanyHoliday = companyHolidayCalendars?.find(
    (companyHoliday) => companyHoliday.holidayDate === workDate
  );

  if (isHoliday || isCompanyHoliday) return "";

  const isChangeRequesting = changeRequests
    ? changeRequests
        .filter((item): item is NonNullable<typeof item> => !!item)
        .filter((item) => !item.completed).length > 0
    : false;

  if (isChangeRequesting) return "申請中";

  switch (dayOfWeek) {
    case "月":
    case "火":
    case "水":
    case "木":
    case "金":
      if (today === workDate) {
        if (!startTime) return "遅刻";
        if (!endTime) return "勤務中";
      }
      return !startTime || !endTime ? "エラー" : "OK";

    case "土":
    case "日":
      if (!startTime && !endTime) return "";
      return "OK";

    default:
      return "";
  }
}
