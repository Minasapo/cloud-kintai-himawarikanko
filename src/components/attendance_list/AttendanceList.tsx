import { Box, LinearProgress, Stack, Typography } from "@mui/material";
import { DataGrid } from "@mui/x-data-grid";
import { Logger } from "aws-amplify";
import dayjs from "dayjs";
import { useEffect } from "react";
import { useAppDispatchV2 } from "../../app/hooks";
import { Attendance } from "../../client";
import { E02001 } from "../../errors";
import useAttendances from "../../hooks/useAttendances/useAttendances";
import useCognitoUser from "../../hooks/useCognitoUser";
import useHolidayCalendars from "../../hooks/useHolidayCalendars/useHolidayCalendars";
import { setSnackbarError } from "../../lib/reducers/snackbarReducer";
import Title from "../Title/Title";
import GetColumns from "./Column";
import getDayOfWeek, { DayOfWeek } from "./getDayOfWeek";

export default function AttendanceTable() {
  const dispatch = useAppDispatchV2();
  const { cognitoUser, loading: cognitoUserLoading } = useCognitoUser();
  const { attendances, getAttendances } = useAttendances();
  const {
    holidayCalendars,
    loading: holidayCalendarLoading,
    error: holidayCalendarError,
  } = useHolidayCalendars();

  const logger = new Logger(
    "AttendanceList",
    process.env.NODE_ENV === "development" ? "DEBUG" : "ERROR"
  );

  useEffect(() => {
    if (!cognitoUser) return;

    getAttendances(cognitoUser.id).catch((error) => {
      logger.debug(error);
      dispatch(setSnackbarError(E02001));
    });
  }, [cognitoUser]);

  if (holidayCalendarLoading || cognitoUserLoading) {
    return <LinearProgress />;
  }

  if (holidayCalendarError) {
    return <div>データ取得中に何らかの問題が発生しました</div>;
  }

  return (
    <Stack spacing={2}>
      <Box>
        <Title text="勤怠一覧" />
      </Box>
      <Box sx={{ px: 5 }}>
        <Typography variant="body1">
          今日から30日前までの勤怠情報を表示しています
        </Typography>
      </Box>
      <Box sx={{ px: 5, pb: 5 }}>
        <DataGrid
          rows={attendances ?? []}
          columns={GetColumns(holidayCalendars)}
          autoHeight
          hideFooter={true}
          getRowId={(row) => row.workDate}
          getRowClassName={(params: {
            row: { workDate: Attendance["work_date"] };
          }) => {
            const today = dayjs().format("YYYY-MM-DD");
            if (params.row.workDate === today) {
              return "super-app-theme--today";
            }

            const isHoliday = holidayCalendars?.find(
              (holidayCalendar) =>
                holidayCalendar.holidayDate === params.row.workDate
            );

            if (isHoliday) {
              return "super-app-theme--sunday";
            }

            const dayOfWeek = getDayOfWeek(params.row.workDate);
            switch (dayOfWeek) {
              case DayOfWeek.Sat:
                return "super-app-theme--saturday";
              case DayOfWeek.Sun:
                return "super-app-theme--sunday";
              default:
                return "super-app-theme--default";
            }
          }}
          sx={{
            "& .super-app-theme--saturday": {
              backgroundColor: "#93FFFF",
            },
            "& .super-app-theme--sunday": {
              backgroundColor: "#FF9393",
            },
            "& .super-app-theme--today": {
              backgroundColor: "#FFFF93",
              fontWeight: "bold",
            },
          }}
        />
      </Box>
    </Stack>
  );
}
