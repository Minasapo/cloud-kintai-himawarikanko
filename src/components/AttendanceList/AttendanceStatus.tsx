import CheckCircleIcon from "@mui/icons-material/CheckCircle";
import ErrorIcon from "@mui/icons-material/Error";
import { Box, Tooltip } from "@mui/material";

import {
  Attendance,
  CompanyHolidayCalendar,
  HolidayCalendar,
  Staff,
} from "../../API";
import { judgeStatus } from "./common";

export function AttendanceStatus({
  staff,
  attendance,
  holidayCalendars,
  companyHolidayCalendars,
}: {
  staff: Staff | null | undefined;
  attendance: Attendance;
  holidayCalendars: HolidayCalendar[];
  companyHolidayCalendars: CompanyHolidayCalendar[];
}) {
  const { workDate, startTime, endTime, paidHolidayFlag, changeRequests } =
    attendance;

  const getStatus = judgeStatus(
    workDate,
    startTime,
    endTime,
    holidayCalendars,
    companyHolidayCalendars,
    paidHolidayFlag,
    changeRequests,
    staff
  );

  if (getStatus === "") return <Box width={24} height={24} />;

  return getStatus === "OK" ? (
    <CheckCircleIcon color="success" />
  ) : (
    <Tooltip title="勤怠に不備があります">
      <ErrorIcon color="error" />
    </Tooltip>
  );
}
