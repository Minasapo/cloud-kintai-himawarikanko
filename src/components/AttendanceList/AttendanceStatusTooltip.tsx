import CheckCircleIcon from "@mui/icons-material/CheckCircle";
import ErrorIcon from "@mui/icons-material/Error";
import HourglassTopIcon from "@mui/icons-material/HourglassTop";
import { Box, Tooltip } from "@mui/material";

import {
  Attendance,
  CompanyHolidayCalendar,
  HolidayCalendar,
  Staff,
} from "../../API";
import { judgeStatus } from "./common";

export function AttendanceStatusTooltip({
  staff,
  attendance,
  holidayCalendars,
  companyHolidayCalendars,
}: {
  staff: Staff | null | undefined;
  attendance: Attendance;
  holidayCalendars: HolidayCalendar[];
  companyHolidayCalendars: CompanyHolidayCalendar[];
}) {
  const {
    workDate,
    startTime,
    endTime,
    paidHolidayFlag,
    changeRequests,
    substituteHolidayDate,
  } = attendance;

  const getStatus = judgeStatus(
    workDate,
    startTime,
    endTime,
    holidayCalendars,
    companyHolidayCalendars,
    paidHolidayFlag,
    changeRequests,
    staff,
    substituteHolidayDate
  );

  if (getStatus === "") return <Box width={24} height={24} />;

  switch (getStatus) {
    case "OK":
    case "勤務中":
      return <CheckCircleIcon color="success" />;
    case "申請中":
      return (
        <Tooltip title="申請中です。承認されるまで反映されません">
          <HourglassTopIcon color="warning" />
        </Tooltip>
      );
    case "遅刻":
    case "エラー":
      return (
        <Tooltip title="勤怠に不備があります">
          <ErrorIcon color="error" />
        </Tooltip>
      );

    default:
      return <Box width={24} height={24} />;
  }
}
