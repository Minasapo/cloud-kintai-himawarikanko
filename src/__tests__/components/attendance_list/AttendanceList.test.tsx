import { render } from "@testing-library/react";

import AttendanceList from "../../../components/attendance_list/AttendanceList";

describe("AttendanceList Component(", () => {
  test.concurrent("出勤一覧", () => {
    const { asFragment } = render(<AttendanceList />);
    expect(asFragment()).toMatchSnapshot();
  });
});
