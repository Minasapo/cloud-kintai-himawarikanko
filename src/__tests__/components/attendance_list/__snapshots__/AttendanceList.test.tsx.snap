// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<DocumentFragment>
  <div
    style="height: 100vh; width: 100%;"
  >
    <div
      aria-colcount="9"
      aria-multiselectable="false"
      aria-rowcount="1"
      class="MuiDataGrid-root MuiDataGrid-root--densityStandard css-1e2bxag-MuiDataGrid-root"
      role="grid"
    >
      <div>
        <div />
      </div>
      <div
        class="MuiDataGrid-main css-204u17-MuiDataGrid-main"
      >
        <div
          style="height: -56px; width: 0px; position: absolute; top: 56px;"
        >
          <div>
            データがありません
          </div>
        </div>
        <div
          class="MuiDataGrid-columnHeaders css-z8fhq1-MuiDataGrid-columnHeaders"
          style="min-height: 56px; max-height: 56px; line-height: 56px;"
        >
          <div
            class="MuiDataGrid-columnHeadersInner MuiDataGrid-columnHeadersInner--scrollable css-gl260s-MuiDataGrid-columnHeadersInner"
            role="rowgroup"
            style="transform: translate3d(0px, 0px, 0px);"
          >
            <div
              aria-rowindex="1"
              class="css-yrdy0g-MuiDataGrid-columnHeaderRow"
              role="row"
            >
              <div
                aria-colindex="1"
                aria-sort="none"
                class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--alignCenter MuiDataGrid-columnHeaderCheckbox"
                data-field="__check__"
                role="columnheader"
                style="height: 56px; width: 50px; min-width: 50px; max-width: 50px;"
                tabindex="0"
              >
                <div
                  class="MuiDataGrid-columnHeaderDraggableContainer"
                  draggable="false"
                >
                  <div
                    class="MuiDataGrid-columnHeaderTitleContainer"
                  >
                    <div
                      class="MuiDataGrid-columnHeaderTitleContainerContent"
                    >
                      <span
                        class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary MuiDataGrid-checkboxInput css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                      >
                        <input
                          aria-label="Select all rows"
                          class="PrivateSwitchBase-input css-1m9pwf3"
                          data-indeterminate="false"
                          tabindex="-1"
                          type="checkbox"
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="CheckBoxOutlineBlankIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </span>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--sideRight"
                  style="min-height: 56px; opacity: 1;"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                    data-testid="SeparatorIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M11 19V5h2v14z"
                    />
                  </svg>
                </div>
              </div>
              <div
                aria-colindex="2"
                aria-label="名前"
                aria-sort="none"
                class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--alignCenter"
                data-field="staffName"
                role="columnheader"
                style="height: 56px; width: 100px; min-width: 100px; max-width: 100px;"
                tabindex="-1"
              >
                <div
                  class="MuiDataGrid-columnHeaderDraggableContainer"
                  draggable="false"
                >
                  <div
                    class="MuiDataGrid-columnHeaderTitleContainer"
                  >
                    <div
                      class="MuiDataGrid-columnHeaderTitleContainerContent"
                    >
                      <div
                        aria-label=""
                        class="MuiDataGrid-columnHeaderTitle css-1jbbcbn-MuiDataGrid-columnHeaderTitle"
                        data-mui-internal-clone-element="true"
                      >
                        名前
                      </div>
                    </div>
                  </div>
                  <div
                    class="MuiDataGrid-menuIcon"
                  >
                    <button
                      aria-controls=":r2:"
                      aria-haspopup="true"
                      aria-label="Menu"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall MuiDataGrid-menuIconButton css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                      id=":r3:"
                      tabindex="-1"
                      title="Menu"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                        data-testid="TripleDotsVerticalIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
                <div
                  class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--sideRight"
                  style="min-height: 56px; opacity: 1;"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                    data-testid="SeparatorIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M11 19V5h2v14z"
                    />
                  </svg>
                </div>
              </div>
              <div
                aria-colindex="3"
                aria-label="ステータス"
                aria-sort="none"
                class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--alignCenter"
                data-field="status"
                role="columnheader"
                style="height: 56px; width: 100px; min-width: 100px; max-width: 100px;"
                tabindex="-1"
              >
                <div
                  class="MuiDataGrid-columnHeaderDraggableContainer"
                  draggable="false"
                >
                  <div
                    class="MuiDataGrid-columnHeaderTitleContainer"
                  >
                    <div
                      class="MuiDataGrid-columnHeaderTitleContainerContent"
                    >
                      <div
                        aria-label=""
                        class="MuiDataGrid-columnHeaderTitle css-1jbbcbn-MuiDataGrid-columnHeaderTitle"
                        data-mui-internal-clone-element="true"
                      >
                        ステータス
                      </div>
                    </div>
                  </div>
                  <div
                    class="MuiDataGrid-menuIcon"
                  >
                    <button
                      aria-controls=":r5:"
                      aria-haspopup="true"
                      aria-label="Menu"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall MuiDataGrid-menuIconButton css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                      id=":r6:"
                      tabindex="-1"
                      title="Menu"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                        data-testid="TripleDotsVerticalIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
                <div
                  class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--sideRight"
                  style="min-height: 56px; opacity: 1;"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                    data-testid="SeparatorIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M11 19V5h2v14z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          style="overflow: visible; height: 0px; width: 0px;"
        >
          <div
            class="MuiDataGrid-virtualScroller css-1w5m2wr-MuiDataGrid-virtualScroller"
            style="width: 0px; height: auto; margin-top: 56px; overflow-x: hidden;"
          >
            <div
              class="MuiDataGrid-virtualScrollerContent MuiDataGrid-virtualScrollerContent--overflowed css-1kwdphh-MuiDataGrid-virtualScrollerContent"
              style="width: auto; height: 1px; min-height: auto;"
            >
              <div
                class="MuiDataGrid-virtualScrollerRenderZone css-s1v7zr-MuiDataGrid-virtualScrollerRenderZone"
                style="transform: translate3d(0px, undefinedpx, 0px);"
              />
            </div>
          </div>
        </div>
        <div
          class="Mui-resizeTriggers"
        >
          <div
            class="expand-trigger"
          >
            <div
              style="width: 1px; height: 1px;"
            />
          </div>
          <div
            class="contract-trigger"
          />
        </div>
      </div>
      <div />
    </div>
  </div>
</DocumentFragment>
`;
